app/components/provider_interface/status_box_component.rb

```diff
@@ -4,8 +4,10 @@ class StatusBoxComponent < ActionView::Component::Base
 
     attr_reader :application_choice
 
-    def initialize(application_choice:)
+    def initialize(application_choice:, extra_arguments: nil)
```

Comment:
I think the `StatusBoxComponent` pattern is a bit compromised now, not necessarily as part of this PR but because status boxes are conditionally rendered. Probably not a change needed in this PR but we have 2 cases of not rendering the box depending on status and this PR introduces `extra_arguments` to satisfy another piece of conditional logic.
Something to look at in the future, I think we've moved away from always rendering something based on `ApplicationChoice#status` and into some bespoke renderings.
Maybe that points towards different views or partials to satisfy this portion of the page as a one size fits all component seems to be a stretch now.

Reply:
I agree the more conditional we make these renderings the more a different approach may be called for. Allowing the passing of optional bits of information is necessarily a bad thing, though. In fact, if we stick with the current pattern for now, Duncan suggested we go all the way and just pass a hash to the top-level initializer, then have the subcomponents complain if they don't get what they need. I think I'll just do that for now. Maybe something to discuss at tech retro?